### PR TITLE
Row action automation display in frontend

### PR DESCRIPTION
**Pull request by adrinr**
## Description
Displaying row action information, such as table name next to action name and trigger information banners. This required changing the fetch automation endpoint in order to make it more extensible without polluting the database objects or having copies of reference names

## Addresses
- [BUDI-8430 - Row action API - auto-create automation
](https://linear.app/budibase/issue/BUDI-8430/row-action-api-auto-create-automation)

## Screenshots
image

### CODE REVIEW

File: packages/builder/src/components/automation/AutomationBuilder/FlowChart/TestDataModal.svelte [hunk 1]
```diff
@@ -81,7 +81,7 @@
   // Check the schema to see if required fields have been entered
   $: isError =
     !isTriggerValid(trigger) ||
-    !trigger.schema.outputs.required?.every(
+    !(trigger.schema.outputs.required || []).every(
```
Comment (mike12345567) [inline packages/builder/src/components/automation/AutomationBuilder/FlowChart/TestDataModal.svelte:84]: Why is this change needed?

File: packages/frontend-core/src/api/automations.js [hunk 2]
```diff
@@ -26,9 +26,13 @@ export const buildAutomationEndpoints = API => ({
   /**
    * Gets a list of all automations.
    */
-  getAutomations: async () => {
+  getAutomations: async ({ enrich }) => {
+    let url = "/api/automations?"
+    if (enrich) {
+      url += "enrich=true"
```
Comment (mike12345567) [inline packages/frontend-core/src/api/automations.js:32]: Can we use `URLSearchParams` here to add this to the URL? Helpful if this ever needs to be extended with other parameters.
e.g.
```
const params = new URLSearchParams()
if (enrich) {
  params.set("enrich", true)
}
const url = `/api/automations?${params.toString()}`
```

File: packages/server/src/api/controllers/automation.ts [hunk 3]
```diff
@@ -73,8 +74,17 @@ export async function update(ctx: UserCtx) {
   builderSocket?.emitAutomationUpdate(ctx, automation)
 }
 
-export async function fetch(ctx: UserCtx) {
-  ctx.body = await sdk.automations.fetch()
+export async function fetch(ctx: UserCtx<void, FetchAutomationResponse>) {
+  const query = ctx.request.query || {}
```
Comment (mike12345567) [inline packages/server/src/api/controllers/automation.ts:78]: Might be nice to type this, even inline

**Review by mike12345567 — APPROVED** [review]
LGTM - some NAB comments just!

File: packages/builder/src/components/automation/AutomationBuilder/FlowChart/TestDataModal.svelte (hunk 1 above)
Comment (adrinr) [reply packages/builder/src/components/automation/AutomationBuilder/FlowChart/TestDataModal.svelte:84]: If we don't have any required field this will always be true, as `!undefined`. So, not having required fields, will always result on `isError = true`

Comment (mike12345567) [reply packages/builder/src/components/automation/AutomationBuilder/FlowChart/TestDataModal.svelte:84]: Ah makes sense!

**Review by mike12345567 — APPROVED** [review]
